File: ollama/codigo-final/fasta/qwen2.5-coder-7b/fewshot/temp_0/fasta.java
import java.util.Random;

public class DNASequenceGenerator {
    private static final String ALU_SEQUENCE = "GGCCGGGCGCGGTGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGCGGGCGGA" +
            "TCACCTGAGGTCAGGAGTTCGAGACCAGCCTGGCCAACATGGTGAAACCCCGTCTCTACT" +
            "AAAAATACAAAAATTAGCCGGGCGTGGTGGCGCGCGCCTGTAATCCCAGCTACTCGGGAG" +
            "GCTGAGGCAGGAGAATCGCTTGAACCCGGGAGGCGGAGGTTGCAGTGAGCCGAGATCGCG" +
            "CCACTGCACTCCAGCCTGGGCGACAGAGCGAGACTCCGTCTCAAAAA";
    private static final double[] IUB_PROBABILITIES = {0.27, 0.12, 0.12, 0.27, 0.02, 0.02, 0.02, 0.02, 0.02, 0.02, 0.02, 0.02};
    private static final double[] HOMO_SAPIENS_PROBABILITIES = {0.3029549426680, 0.1979883004921, 0.1975473066391, 0.3015094502008};
    private static final int IM = 139968;
    private static final int IA = 3877;
    private static final int IC = 29573;
    private static final int SEED = 42;

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java DNASequenceGenerator <N>");
            return;
        }

        int N = Integer.parseInt(args[0]);
        Random random = new Random(SEED);

        // Generate ALU sequence
        StringBuilder aluSequence = new StringBuilder();
        for (int i = 0; i < N; ) {
            int index = random.nextInt(ALU_SEQUENCE.length());
            char nucleotide = ALU_SEQUENCE.charAt(index);
            if (nucleotide != ' ') { // Ignore spaces in the ALU sequence
                aluSequence.append(nucleotide);
                i++;
            }
        }

        // Generate IUB sequence
        StringBuilder iubSequence = new StringBuilder();
        for (int i = 0; i < N; ) {
            double randomValue = random.nextDouble();
            int index = 0;
            while (randomValue > IUB_PROBABILITIES[index]) {
                randomValue -= IUB_PROBABILITIES[index];
                index++;
            }
            char nucleotide = "ACGTBDHKMNRYSVWY".charAt(index);
            iubSequence.append(nucleotide);
        }

        // Generate Homo sapiens sequence
        StringBuilder homoSapiensSequence = new StringBuilder();
        for (int i = 0; i < N; ) {
            double randomValue = random.nextDouble();
            int index = 0;
            while (randomValue > HOMO_SAPIENS_PROBABILITIES[index]) {
                randomValue -= HOMO_SAPIENS_PROBABILITIES[index];
                index++;
            }
            char nucleotide = "ACGT".charAt(index);
            homoSapiensSequence.append(nucleotide);
        }

        // Output sequences
        System.out.println(">ALU");
        for (int i = 0; i < aluSequence.length(); i += 60) {
            System.out.println(aluSequence.substring(i, Math.min(i + 60, aluSequence.length())));
        }
        System.out.println(">IUB");
        for (int i = 0; i < iubSequence.length(); i += 60) {
            System.out.println(iubSequence.substring(i, Math.min(i + 60, iubSequence.length())));
        }
        System.out.println(">Homo sapiens");
        for (int i = 0; i < homoSapiensSequence.length(); i += 60) {
            System.out.println(homoSapiensSequence.substring(i, Math.min(i + 60, homoSapiensSequence.length())));
        }
    }
}

File: ollama/codigo-final/fasta/qwen2.5-coder-7b/fewshot/temp_0.7/fasta.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cstdlib>

// LCG Constants
const int IM = 139968;
const int IA = 3877;
const int IC = 29573;

// Seed for the random number generator
int seed = 42;

// Function to generate a random number using LCG
int lcg() {
    seed = (seed * IA + IC) % IM;
    return seed;
}

// Function to generate DNA sequence based on probabilities
std::string generateDnaSequence(int length, const std::vector<std::pair<char, double>>& alphabet, int totalProbability) {
    std::string sequence;
    for (int i = 0; i < length; ++i) {
        double randomValue = lcg() / static_cast<double>(IM);
        double cumulativeProbability = 0.0;
        for (const auto& entry : alphabet) {
            cumulativeProbability += entry.second;
            if (randomValue <= cumulativeProbability) {
                sequence += entry.first;
                break;
            }
        }
    }
    return sequence;
}

int main(int argc, char* argv[]) {
    if (argc != 2) {
        std::cerr << "Usage: " << argv[0] << " <N>" << std::endl;
        return 1;
    }

    int N = std::atoi(argv[1]);

    // ALU sequence
    const std::string aluSequence("GGCCGGGCGCGGTGGCTCACGCCTGTAATCCCAGCACTTTGGGAGGCCGAGGCGGGCGGA"
                                   "TCACCTGAGGTCAGGAGTTCGAGACCAGCCTGGCCAACATGGTGAAACCCCGTCTCTACT"
                                   "AAAAATACAAAAATTAGCCGGGCGTGGTGGCGCGCGCCTGTAATCCCAGCTACTCGGGAG"
                                   "GCTGAGGCAGGAGAATCGCTTGAACCCGGGAGGCGGAGGTTGCAGTGAGCCGAGATCGCG"
                                   "CCACTGCACTCCAGCCTGGGCGACAGAGCGAGACTCCGTCTCAAAAA");

    // IUB Alphabet and Probabilities
    std::vector<std::pair<char, double>> iubAlphabet = {
        {'a', 0.27}, {'c', 0.12}, {'g', 0.12}, {'t', 0.27},
        {'B', 0.02}, {'D', 0.02}, {'H', 0.02}, {'K', 0.02},
        {'M', 0.02}, {'N', 0.02}, {'R', 0.02}, {'S', 0.02},
        {'V', 0.02}, {'W', 0.02}, {'Y', 0.02}
    };
    int totalIubProbability = 100; // Total probability in percentage

    // Homo sapiens Alphabet and Probabilities
    std::vector<std::pair<char, double>> homoSapiensAlphabet = {
        {'a', 0.3029549426680}, {'c', 0.1979883004921},
        {'g', 0.1975473066391}, {'t', 0.3015094502008}
    };
    int totalHomoSapiensProbability = 100; // Total probability in percentage

    // Generate DNA sequences
    std::string aluSequenceResult = generateDnaSequence(N, {}, N); // ALU sequence is not based on probabilities
    std::string iubResult = generateDnaSequence(N, iubAlphabet, totalIubProbability);
    std::string homoSapiensResult = generateDnaSequence(N, homoSapiensAlphabet, totalHomoSapiensProbability);

    // Output the results
    std::cout << "alu\n";
    for (int i = 0; i < N / 60 + 1; ++i) {
        std::cout << aluSequence.substr(i * 60, 60) << "\n";
    }

    std::cout << "iub\n";
    for (int i = 0; i < N / 60 + 1; ++i) {
        std::cout << iubResult.substr(i * 60, 60) << "\n";
    }

    std::cout << "homosapiens\n";
    for (int i = 0; i < N / 60 + 1; ++i) {
        std::cout << homoSapiensResult.substr(i * 60, 60) << "\n";
    }

    return 0;
}
